### PR TITLE
refactor: encapsulate logic into ClosableBanner

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Follow up https://github.com/cowprotocol/cowswap/pull/4095

Created a new helper `ClosableBanner` to wrap closable (with localStorage flag) banners.

`ClosableBanner` is just a wrapper around `<ClosableBannerInner>` component, for me it looks more neat:

```
ClosableBanner('my-key', (close ) => (
    <Content>
        abc
        <button onClick={close}>close</button>
    </Content>
))
```

Instead of:

```

<ClosableBanner
    storageKey="myKey"
    callback={(close ) => (
        <Content>
            abc
            <button onClick={close}>close</button>
        </Content>
)} />
```

image

  # To Test

Those two banner should work as before

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/limitOrders/pure/InfoBanner/index.tsx [hunk 1]
```diff
@@ -1,45 +1,33 @@
...
-    localStorage.setItem(LOCAL_STORAGE_KEY, 'false')
-  }
-
-  useEffect(() => {
-    localStorage.setItem(LOCAL_STORAGE_KEY, showBanner.toString())
-  }, [showBanner])
-
-  // If the banner is not shown, don't render anything
-  if (!showBanner) return null
+const LOCAL_STORAGE_KEY = 'limitOrders_showInfoBanner'
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/limitOrders/pure/InfoBanner/index.tsx:9]: Haven't changed the value to not reset the flag in the existing users state

File: libs/ui/src/pure/ClosableBanner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+export interface ClosableBannerProps {
+  storageKey: string
+  callback: ClosableBannerCallback
+}
+
+const DEFAULT_STATE: ClosableBannersState = {}
+
+const closableBannersStateAtom = atomWithStorage<ClosableBannersState>(
+  'closable-banners',
```
Comment (alfetopito) [inline libs/ui/src/pure/ClosableBanner/index.tsx:21]: Should this be versioned as well? and camelCase to follow the pattern of other keys?

```suggestion
  'closableBanners:v0',
```

**Review by alfetopito — APPROVED** [review]
Amazing, thanks a lot!

Only one comment

File: libs/ui/src/pure/ClosableBanner/index.tsx (hunk 2 above)
Comment (shoom3301) [reply libs/ui/src/pure/ClosableBanner/index.tsx]: Fixed

**Review by elena-zh — APPROVED** [review]

File: libs/ui/src/pure/ClosableBanner/index.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { useCallback } from 'react'
+
+// true - when banner closed
+type ClosableBannersState = Record<string, true | undefined>
+
+interface ClosableBannerCallback {
+  (close: () => void): JSX.Element
+}
+
+export interface ClosableBannerProps {
```
Comment (anxolin) [inline libs/ui/src/pure/ClosableBanner/index.tsx:13]: the name felt a bit generic for being joitai and localstorage dependant

**Pull request merged**